src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+            }
+            return Optional.empty();
+        } catch (SQLException e) {
+            throw new DataProcessingException("No manufacturer with id: " + id, e);
+        }
+    }
+
+    @Override
+    public List<Manufacturer> getAll() {
+        String getAllQuery = "SELECT * FROM manufacturers WHERE is_deleted = false";
+        List<Manufacturer> allManufacturers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getManufacturesStatement = connection.createStatement()) {
+            ResultSet resultSet = getManufacturesStatement
+                    .executeQuery(getAllQuery);
+            while (resultSet.next()) {
+                allManufacturers.add(getManufacturer(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get all manufacturers from DB", e);
+        }
+        return allManufacturers;
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        String updateQuery = "UPDATE manufacturers SET name = ?, country = ? "
+                + "WHERE id = ? AND is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                   PreparedStatement updateStatement = connection.prepareStatement(updateQuery)) {
+            updateStatement.setString(1, manufacturer.getName());
+            updateStatement.setString(2, manufacturer.getCountry());
+            updateStatement.setLong(3, manufacturer.getId());
+            updateStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to update manufacturer with id: "
+                    + manufacturer.getId(), e);
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE manufacturers SET is_deleted = true WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteStatement = connection
+                           .prepareStatement(deleteRequest)) {
+            deleteStatement.setLong(1, id);
+            return deleteStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to delete the manufacturer with id: "
+                    + id, e);
+        }
+    }
+
+    private Manufacturer getManufacturer(ResultSet resultSet) {
+        try {
+            Manufacturer manufacturer = new Manufacturer();
+            manufacturer.setId(resultSet.getObject("id", Long.class));
+            manufacturer.setName(resultSet.getString("name"));
+            manufacturer.setCountry(resultSet.getString("country"));
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Failed to get manufacturer", e);
```

Comment:
```suggestion
            throw new DataProcessingException("Failed to get manufacturer from ResultSet", e);
```
since we have get method, it's better to add some clarification